staging/src/k8s.io/cli-runtime/pkg/resource/query_param_verifier.go

```diff
@@ -0,0 +1,170 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resource
+
+import (
+	"errors"
+	"fmt"
+
+	openapi_v2 "github.com/googleapis/gnostic/openapiv2"
+	yaml "gopkg.in/yaml.v2"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/dynamic"
+)
+
+func NewQueryParamVerifier(dynamicClient dynamic.Interface, openAPIGetter discovery.OpenAPISchemaInterface) *QueryParamVerifier {
```

Comment:
I think if you had done that, it has a couple of consequences on that PR:
1. You don't have to rename `dryRunVerifier` everywhere
2. You don't have to change "HasSupport" calls anywhere
3. You just need to change the instantiation of the dryRunVerifier into a QueryParamVerifier.

I guess the question is, how is the object most likely to be used. Also, I'm hoping the `openAPIGetter` is cached behind a `sync.Once` if we want to make sure we don't fetch the openapi multiple times.

From how I can see this class being used, I think it's fine if we have two different instances of a QueryParamVerifier, one for dryRun and one for validation.

Reply:
Yeah, I agree. If you're wanting the instance to be shared so it can reuse a looked-up schema, you can have a looker-upper type that does lookup and takes a param as an arg, and another type that embeds that and a specific query param, then the dryRun and fieldValidation verifiers are two instances of that specific-param-support type that can share a looker-upper instance for schema caching